### PR TITLE
Add paper converter

### PR DESCRIPTION
**Pull request by nbirillo**
Add a base converter for the algorithm for detections issues in templates

### CODE REVIEW

File: analysis/src/python/data_analysis/template_errors/postprocessing.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        {SubmissionColumns.STEP_ID.value: 'first', SubmissionColumns.RAW_ISSUE_CLASS.value: 'first',
+         SubmissionColumns.FREQUENCY.value: 'first', SubmissionColumns.POS_IN_TEMPLATE.value: ', '.join})
+    return merged_pos.append(filtered_with_none_position).reset_index(drop=True)
+
+
+def filter_by_freq(
+        templates_df: pd.DataFrame,
+        max_template_freq: int,
+) -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame]:
+    lowest_threshold = 0.1
```
Comment (tiginamaria) [inline analysis/src/python/data_analysis/template_errors/postprocessing.py:57]: Maybe to set by console parameters with default values?

File: analysis/src/python/data_analysis/template_errors/postprocessing.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    return template_issues_df
+
+
+def add_additional_info(
+        template_issues_df: pd.DataFrame,
+        output_path: str,
+        raw_issues_df: Optional[pd.DataFrame] = None,
+) -> pd.DataFrame:
+    template_issues_df['task_link'] = template_issues_df.apply(
+        lambda row: f'https://hyperskill.org/learn/step/{row[SubmissionColumns.STEP_ID.value]}', axis=1)
```
Comment (tiginamaria) [inline analysis/src/python/data_analysis/template_errors/postprocessing.py:116]: Maybe somehow get from dataset with steps of unify with [`analysis/src/python/data_collection/hyperskill/api/steps.py`](https://github.com/hyperskill/hyperstyle-analyze/blob/21db5d9d14e8e83d946f2d81c8249c6561d005ea/analysis/src/python/data_collection/hyperskill/api/steps.py#L103)

File: analysis/src/python/data_analysis/template_errors/postprocessing.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    template_issues_df[description] = final_descriptions
+    return template_issues_df
+
+
+def add_additional_info(
+        template_issues_df: pd.DataFrame,
+        output_path: str,
+        raw_issues_df: Optional[pd.DataFrame] = None,
+) -> pd.DataFrame:
+    template_issues_df['task_link'] = template_issues_df.apply(
```
Comment (tiginamaria) [inline analysis/src/python/data_analysis/template_errors/postprocessing.py:115]: There is already name for such column `analysis.src.python.data_analysis.model.column_name.StepColumns.URL`

File: analysis/src/python/data_analysis/template_errors/postprocessing.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+def save_results(
+        output_path: str,
+        template_issues: pd.DataFrame,
+        rare_typical_issues: pd.DataFrame,
+        common_typical_issues: pd.DataFrame,
+):
+    base_path = f'{output_path}/issues'
+    create_directory(base_path)
+    write_df(template_issues, f'{base_path}/template_issues.csv')
```
Comment (tiginamaria) [inline analysis/src/python/data_analysis/template_errors/postprocessing.py:139]: Maybe usage of `Path` and `AnalysisExtension.CSV` will make this code better

**Review by tiginamaria — CHANGES_REQUESTED** [review]
Greate! Thank you! Fills like you are very tired =( Probably I can fix all comments if you want, because they are only about style, not idea. Also, I think I should go into details of all of this module and write some tests before giving it for industrial usage?

**Comment by nbirillo** [comment]
I will fix it today and add tests, thank you! But first, I need to complete the analytics part for the paper...

**Review by tiginamaria — APPROVED** [review]
Looks perfect! Thanks!

**Pull request merged**